source/kraken/maintenance_worker.cpp

```diff
@@ -176,6 +175,23 @@ void MaintenanceWorker::load_data() {
     LOG4CPLUS_INFO(logger, "Loading database duration: " << duration);
 }
 
+std::pair<uint32_t, uint32_t> MaintenanceWorker::declare_queue_with_counts(const std::string& queue_name) {
```

Comment:
```suggestion
// This function could be called multiple times without creating new queues. The goal is to take advantage 
// of DeclareQueueWithCounts to get the number of messages remaining in the queue currently  
std::pair<uint32_t, uint32_t> MaintenanceWorker::declare_queue_with_counts(const std::string& queue_name) {
```

Reply:
Done in https://github.com/hove-io/navitia/pull/3968/commits/868fc7b3fe6b5012a39fd6e4ae3da3d0e85a022f